staging/src/k8s.io/client-go/dynamic/dynamicinformer/informer.go

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dynamicinformer
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/dynamic/dynamiclister"
+	"k8s.io/client-go/tools/cache"
+	"sync"
+	"time"
+)
+
+// NewDynamicSharedInformerFactory constructs a new instance of dynamicSharedInformerFactory for all namespaces.
+func NewDynamicSharedInformerFactory(client dynamic.Interface, defaultResync time.Duration) DynamicSharedInformerFactory {
+	return &dynamicSharedInformerFactory{
+		client: client,
+		defaultResync:    defaultResync,
+		informers: map[string]DynamicInformer{},
+	}
+}
+
+type dynamicSharedInformerFactory struct {
+	client    dynamic.Interface
+	defaultResync    time.Duration
+
+	lock             sync.Mutex
+	informers map[string]DynamicInformer
+}
+
+var _ DynamicSharedInformerFactory = &dynamicSharedInformerFactory{}
+
+func (f *dynamicSharedInformerFactory)  ForResource(gvr schema.GroupVersionResource, namespace string) DynamicInformer {
```

Comment:
I was hoping that `ForResource` would match the `client-go/informers/factory.go|ForResource(resource schema.GroupVersionResource) (GenericInformer, error)` method.

Reply:
`GenericInformer` returns `GenericLister` which defines `List(selector labels.Selector) (ret []runtime.Object, err error)` whereas our new Lister has `List(selector labels.Selector) (ret []*unstructured.Unstructured, err error)`

Unfortunately the golang compiler will not automatically convert `ret []runtime.Object` to `ret []*unstructured.Unstructured`.

Reply:
sgtm.